src/documentation/Search.tsx

```diff
@@ -30,15 +31,9 @@ const Search = ({
   query,
   setQuery,
   onQueryChange,
+  onClear,
 }: SearchProps) => {
   const ref = useRef<HTMLInputElement>(null);
-  const handleClear = useCallback(() => {
-    setQuery("");
-    setResults(undefined);
-    if (ref.current) {
-      ref.current.focus();
-    }
-  }, [setQuery, setResults]);
```

Comment:
Let's try keeping this to do the focus handling then call onClear for the parent state to update.